src/core/unit.c

```diff
@@ -5450,7 +5450,13 @@ int unit_prepare_exec(Unit *u) {
 
         /* Prepares everything so that we can fork of a process for this unit */
 
-        (void) unit_realize_cgroup(u);
+        r = unit_realize_cgroup(u);
+        if (r < 0 && IN_SET(u->type, UNIT_MOUNT, UNIT_SWAP)) {
+                /* Realizing the cgroup failed, but as a fallback we will run mount/unmount or swapon/swapoff in "init.scope". */
+
+                (void) unit_set_cgroup_path(u, "/init.scope");
```

Comment:
Macro `SPECIAL_INIT_SCOPE` would be better here.

But I'm worried `unit_realize_cgroup()` is not ready for such (ab)use (it'd unnecessarily enable controllers in regular tree, it'd modify attributes of `init.scope` unexpectedly).
The simpler approach could be to zero `exec_params.cgroup_path` so that the helper commands forked by PID 1 just remain in its own cgroup.

Reply:
I will experiment with this alternative approach, it might be a bit safer.

Reply:
I just had a closer look at `unit_set_cgroup_path()` and you are absolutely right. I can't do this, because otherwise I'd remove the association between `init.scope` and unit object for that unit and I'd replace it with unit object of mount/swap unit.

Reply:
In the latest version I am just unsetting `u->cgroup_path` and that seems to work fine. PTAL